src/TicketingSystem.BusinessLogic/Services/KafkaNotificationService.cs

```diff
@@ -0,0 +1,73 @@
+﻿using com.ticketingSystem;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TicketingSystem.BusinessLogic.Dtos;
+using TicketingSystem.BusinessLogic.Models;
+using TicketingSystem.Common.Enums;
+using TicketingSystem.Messaging.Models.Models;
+using TicketingSystem.Messaging.Producer;
+
+namespace TicketingSystem.BusinessLogic.Services
+{
+    public class KafkaNotificationService(IKafkaProducer kafkaProducer, INotificationService notificationService, IEventService eventService)
+        : IKafkaNotificationService
+    {
+        private readonly IKafkaProducer _producer = kafkaProducer;
+        private readonly INotificationService _notificationService = notificationService;
+        private readonly IEventService _eventService = eventService;
+
+        public Task CreateSeatsBookedNotification(PaymentDto payment, List<EventSectionSeatsModel> groupedCartItems)
+        {
+            return CreateNotification(OperationType.SeatsBooked, payment, groupedCartItems);
+        }
+
+        public Task CreatePaymentSucceededNotification(PaymentDto payment, List<EventSectionSeatsModel> groupedCartItems)
+        {
+            return CreateNotification(OperationType.PaymentSucceeded, payment, groupedCartItems);
+        }
+
+        private async Task CreateNotification(OperationType operation, PaymentDto payment, List<EventSectionSeatsModel> groupedCartItems)
+        {
+            var notificationId = await _notificationService.CreateNotification(payment.Id);
+
+            var seatsAmount = groupedCartItems.Sum(x => x.SectionSeats.Sum(x => x.SeatIds.Length));
+            var seatsInfo = payment.CartItems.Select(x => new
+            {
+                x.EventId,
+                Info = $"{x.EventSectionClass}{x.EventSectionNumber} - seat {x.EventSeatNumber}"
+            });
+
+            var groupedSeats = seatsInfo.GroupBy(x => x.EventId);
+
+            var info = new StringBuilder();
+
+            foreach (var one in groupedSeats)
+            {
+                var ev = await _eventService.GetByIdAsync(one.Key);
```

Comment:
No matter how many events you expect to have in your common application flow - if there is a possibility to have not one but many - it is always better to get them in a bulk and then process instead of one by one call to DB in foreach cycle.